src/v/kafka/server/handlers/fetch.cc

```diff
@@ -737,9 +735,10 @@ class fetch_worker {
     ss::gate _waiter_gate;
     ss::abort_source _as;
 
-    // This signals when an index has been added to `_request_indexes` or when
-    // `_as` aborts.
-    ss::condition_variable _has_completed_waiters;
+    // This should roughly correspond to `_request_indexes.size()`. It's used to
+    // wait for partitions to have an offset change so that it's corresponding
+    // request in the fetch can be re-queried.
+    ss::semaphore _completed_waiter_count{0};
```

Comment:
`ssx::semaphore` not sure it will be much more useful here.

Reply:
We should use it, because the guideline is that all semaphores should be named and I think the semaphores in this change are "important" (there are lot of them, on the hot path, etc).

So please name this semaphore.

Reply:
Good point, will do.